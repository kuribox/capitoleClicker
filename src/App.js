import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import StoreContext from './providers/StoreContext';
import RootStore from './stores/RootStore';

import Routes from './routes';

const rootStore = new RootStore();

function App() {
  return (
    <StoreContext.Provider value={rootStore}>
      <NavigationContainer>
        <Routes />
      </NavigationContainer>
    </StoreContext.Provider>
  );
}

export default App;