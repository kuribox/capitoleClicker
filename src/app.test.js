import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
import MockAsyncStorage from 'mock-async-storage';

const mockImpl = new MockAsyncStorage();

jest.mock('@react-native-async-storage/async-storage', () => mockImpl);
jest.useFakeTimers();

it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});