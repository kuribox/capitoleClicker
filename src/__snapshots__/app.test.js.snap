// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        hideShadow={false}
        largeTitleHideShadow={false}
        title="Homes"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "justifyContent": "space-between",
              "padding": 10,
            }
          }
          style={
            Object {
              "backgroundColor": "white",
              "flexDirection": "column",
              "flexGrow": 1,
              "height": 200,
            }
          }
        >
          <View>
            <Image
              source={1}
              style={
                Object {
                  "maxWidth": "80%",
                  "resizeMode": "contain",
                }
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flex": 1,
                  "justifyContent": "center",
                  "marginBottom": 20,
                  "paddingVertical": 10,
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                  }
                }
              >
                Please enter your username
              </Text>
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                rejectResponderTermination={true}
                style={
                  Object {
                    "borderColor": "#072146",
                    "borderRadius": 5,
                    "borderWidth": 1,
                    "height": 40,
                    "margin": 12,
                    "padding": 10,
                    "width": "100%",
                  }
                }
                testID="user-input"
                underlineColorAndroid="transparent"
                value={null}
              />
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": true,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="login-button"
              >
                <View
                  style={
                    Array [
                      Object {},
                      Object {},
                    ]
                  }
                >
                  <Text
                    disabled={true}
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "margin": 8,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "#072146",
                        },
                        Object {
                          "color": "#cdcdcd",
                        },
                      ]
                    }
                  >
                    Join
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "width": "100%",
                }
              }
            />
          </View>
        </RCTScrollView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
